File: python/Point.py
# Program
# Filename:    Point.py

# Title:		   Point Class (version 1.0)
# Created on: 	July 29, 2020

# Last Date
# Modified:	

# Author:	   Maurice Tedder (based on the Java Applet by James Macgill - http://www.ccg.leeds.ac.uk/james/aStar/

# Target
# Compilers:	python 3

# Description:	Point class indicating grid position coordinaes on the gridmap.

# 
class Point:
   def __init__(self, x=0, y=0):
      self.x = x
      self.y = y

   # display this class as readable string in print methods.
   def __str__(self):
        return '(' + str(self.x) + ',' + str(self.y) + ')'   

File: java/astar/com/bmai/astar/Point.java
package com.bmai.astar;

/**
 * 
 * Program
 * Filename:	Point.java
 * 
 * Title:		Point Class (version 1.0)
 * Created on: 	August 21, 2020
 * 
 * Last Date
 * Modified:	
 *  
 * @author		Maurice Tedder
 * 
 * Target
 * Compilers:	Java - j2sdk 1.4.2
 *
 * Description:	Class that represents a X & Y coordinate point position of a GridCell on a GridMap.
 */
public class Point {
    public int x;
    public int y;

    /**
     * Parameterized constructor.
     * @param x
     * @param y
     */
    public Point(int x, int y){
        this.x = x;
        this.y = y;
    }

    /**
    * Override toString method for this class/object.
    */
    public String toString(){
        return "(" + this.x + "," + this.y + ")";
    }
}